cobigen-cli/cli/src/main/java/com/devonfw/cobigen/cli/commands/GenerateCommand.java

```diff
@@ -243,6 +273,25 @@ private void setRootOutputDirectoryWithPrompt() {
 
   }
 
+  /**
+   * Opens a looping prompt with a yes/no question and upgrades the templates to the newest version.
+   *
+   */
+  private void askUserToUpgradeTemplates() {
+
+    Path outputDirectory = Paths.get(System.getProperty("user.dir"));
+
+    boolean setToUserDir = ValidationUtils.yesNoPrompt("upgrading templates version...: " + outputDirectory.toString(),
+        "Invalid input. Please answer yes/n or no/n (or hit return for yes).",
+        "Continue generation with old templates...");
```

Comment:
Can you make a Screenshot please of the current message flow? Just from what I interpret from the code it is not really clear to the user what needs to be done or what is the reason for the question.

Reply:
![grafik](https://user-images.githubusercontent.com/96188427/176396864-4eb3c621-afd3-49d3-9a85-b39d1ba15fa8.png)